src/utils/wallet.js

```diff
@@ -314,19 +488,46 @@ class Wallet {
         this.accounts[accountId] = true
     }
 
-    async addAccessKey(accountId, contractId, publicKey) {
-        try {
-            return await this.getAccount(accountId).addKey(
-                publicKey,
-                contractId,
-                '', // methodName
-                ACCESS_KEY_FUNDING_AMOUNT
-            )
-        } catch (e) {
-            if (e.type === 'AddKeyAlreadyExists') {
-                return true;
+    async addAccessKey(accountId, contractId, publicKey, fullAccess = false) {
+        const { account, has2fa } = await this.getAccountAndState()
+        if (has2fa) {
+            // default method names will be limited access key multisig contract methods
+            let method_names = METHOD_NAMES_LAK
```

Comment:
wouldn't this path also need handling for `AddKeyAlreadyExists` error?

Reply:
added a preemptive check to twoFactorAddKey in twoFactor.js, the mutlisig request throws the error on confirm from contract helper